### PR TITLE
fix(route): layoffs.fyi

### PR DESCRIPTION
**Pull request by BrandNewLifeJackie26**
## 该 PR 相关 Issue / Involved issue

N/A

## 完整路由地址 / Example for the proposed route(s)

```routes
/layoffs
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [ ] 新的路由 New Route
  - [ ] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [ ] 文档说明 Documentation
  - [ ] 中文文档 CN
  - [ ] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note
Fix 401 error by updating the target URL.

### CODE REVIEW

File: lib/v2/layoffs/index.js [hunk 1]
```diff
@@ -4,22 +4,8 @@ const { parseDate } = require('@/utils/parse-date');
...
-%22%3A%22viwA14Z1pM69YIsaW%22%2C%22action%22%3A%22readSignedAttachmentUrls
-%22%7D%2C%7B%22modelClassName%22%3A%22row%22%2C%22modelIdSelector%22%3A%22
-rows%20*%5BdisplayedInView%3DviwA14Z1pM69YIsaW%5D%22%2C%22action%22%3A%22
-createBoxDocumentSession%22%7D%2C%7B%22modelClassName%22%3A%22row%22%2C%22
-modelIdSelector%22%3A%22rows%20*%5BdisplayedInView%3DviwA14Z1pM69YIsaW%5D%22%2C%22
-action%22%3A%22createDocumentPreviewSession%22%7D%5D%2C%22shareId%22%3A%22shrqYt5kSqMzHV9R5%22%2C%22
-applicationId%22%3A%22app1PaujS9zxVGUZ4%22%2C%22generationNumber%22%3A0%2C%22
-expires%22%3A%222023-02-02T00%3A00%3A00.000Z%22%2C%22
-signature%22%3A%22096478e17705ffdac261da07c5b4c025aff6d8c6d86be027533fd248913798de%22%7D`;
+const SOURCE_URL =
```
Comment (TonyRL) [inline lib/v2/layoffs/index.js:7]: Please parse this URL programmatically. I don't think you like updating it every month.

You can obtain it from `window.__stashedPrefetch.urlWithParams` in `https://airtable.com/embed/shrqYt5kSqMzHV9R5`

**Review by TonyRL — CHANGES_REQUESTED** [review]

File: lib/v2/layoffs/index.js (hunk 1 above)
Comment (BrandNewLifeJackie26) [reply lib/v2/layoffs/index.js:7]: Sounds good, I will post a PR tomorrow, thanks for the advice!!

Comment (BrandNewLifeJackie26) [reply lib/v2/layoffs/index.js]: Added a cache to cache the data source link and re-fetch it when the URL is invalid.

File: lib/v2/layoffs/index.js [hunk 2]
```diff
@@ -57,12 +44,59 @@ module.exports = async (ctx) => {
...
+    } else {
+        // Try to fetch data,
+        // it may fail due to outdated url
+        try {
+            response = await got({
+                method: 'get',
+                url: dataSourceUrl,
+                headers,
+            });
+            if (!response.ok) {
```
Comment (TonyRL) [inline lib/v2/layoffs/index.js:62]: `.ok` is added in got v12, RSSHub uses got v11. You can check the `statusCode` instead.
```js
if (response.statusCode >= 400)
```

File: lib/v2/layoffs/index.js [hunk 3]
```diff
@@ -57,12 +44,59 @@ module.exports = async (ctx) => {
...
+        const SEARCH_STRING = 'urlWithParams: "';
+        for (const scriptObj of $('script').get()) {
+            const scriptText = $(scriptObj).text();
+            const index = scriptText.indexOf(SEARCH_STRING);
+            if (index > 0) {
+                const nextIndex = scriptText.indexOf('"', index + SEARCH_STRING.length);
+                dataSourceUrl = AIRTABLE_HOST + scriptText.substring(index + SEARCH_STRING.length, nextIndex).replaceAll('\\u002F', '/');
+                break;
+            }
+        }
```
Comment (TonyRL) [inline lib/v2/layoffs/index.js:86]:
```suggestion
        dataSourceUrl =
            AIRTABLE_HOST +
            $('script')
                .text()
                .match(/urlWithParams: "(.*?)"/)[1]
                .replace(/\\u002F/g, '/');
```
Note that RSSHub supports down to node v14 which does not have `replaceAll`

File: lib/v2/layoffs/index.js (hunk 2 above)
Comment (BrandNewLifeJackie26) [reply lib/v2/layoffs/index.js]: Fixed, thanks!

File: lib/v2/layoffs/index.js (hunk 3 above)
Comment (BrandNewLifeJackie26) [reply lib/v2/layoffs/index.js]: Thanks for the more elegant solution!

**Pull request merged**